Makefile

```diff
@@ -985,6 +985,28 @@ protos/lint: buf/installed
 .PHONY: lint-protos
 lint-protos: protos/lint
 
+#
+# Runs a check to make sure that the container versions for a target branch are up to date.
+# Useful when ensuring that container versions have been updated properly after cutting a new major version release.
+#
+.PHONY: lint-container-versions
+lint-container-versions:
+	@# We try using the target branch for Github Actions, then Google Cloud Build, and then simply the current git branch if those are empty.
+	@BASE_BRANCH="$${GITHUB_BASE_REF:-$${_BASE_BRANCH:-$$(git rev-parse --abbrev-ref HEAD)}}"; \
+	if echo $${BASE_BRANCH} | grep -qE ".*branch\/v[0-9]+$$"; then \
+		VERSION_NUMBER="$$(echo $${BASE_BRANCH} | sed "s/.*branch\/v//g")"; \
+		OUTPUT="$$(git grep "public.ecr.aws/gravitational/teleport[^:]*:" | grep -E "^[^\.].*?\.ya?ml:" | grep -v "/tests" | grep -v "\/teleport[^:]*:$${VERSION_NUMBER}.*\$$" )"; \
```

Comment:
Just figured I'd explain this mess:

* `git grep "public.ecr.aws/gravitational/teleport[^:]*:"` : Anything with `public.ecr.aws/gravitational/teleport` in it ending in a `:` character is grepped for. Requiring the ending `:` avoids the `values.yaml` files in the helm charts present in the `examples/chart` directories.
* `grep -E "^[^\.].*?\.ya?ml:"` Get any `yaml` or `yml` file that doesn't begin with a `.`, excluding things like .`drone.yml` and `.cloudbuild`. Thinking on this, we may want to include `.cloudbuild`.
* `grep -v "/tests"` removes anything in a test directory. In particular, this removes the `*_test.yaml.snap` files present in the helm charts.
* `grep -v "\/teleport[^:]*:${VERSION_NUMBER}.*\$"` removes any line that has the expected version number. This is somewhat duplicative, but is done to avoid scenarios where the `:` characters at the end of a grepped line might falsely trip up this command.

This whole command is then tested for zero length. If the output is empty, that means there are no lines that have a mismatched version. Otherwise the error is printed and the script in the `Makefile` exits abnormally.